scipy/stats/_qmc_cy.pyx

```diff
@@ -0,0 +1,326 @@
+# cython: language_level=3
+# cython: boundscheck=False
+# cython: wraparound=False
+# cython: cdivision=True
+
+# distutils: language = c++
+
+import numpy as np
+cimport numpy as np
+from libc.math cimport fabs, sqrt, pow
+
+np.import_array()
+
+cdef extern from "<thread>" namespace "std" nogil:
+    cdef cppclass thread:
+        thread()
+        void thread[A, B, C, D, E, F](A, B, C, D, E, F)
```

Comment:
Maybe all these could benefit from comments.

Reply:
In C++, thread constructor can be call with any (≥2) number of args, with any type, it is fully templated. This is defined in `<thread>`. But cython need to know that thread is a class which accept 6 args (because we use 6), this have no effect on C++ code, but allow cython to convert the code to C++. It is why I did that in my test (this is a little bit dirty, I think, but the only method I found). @V0lantis use the same trick in his code. If you think it is dirty, I agree, and it is my fault.

Reply:
By all mean I have no opinion about whether this is dirty of not as I am not a c++ dev 😅 This looked just not obvious for someone like me and I thought maybe we could comment this so that non c++ dev could still go around.

Reply:
Will do this